### PR TITLE
Refresh all CountryActivityLogs per hour

### LINKED ISSUE #3161: Heroku PG OutOfMemory error
Heroku log:

```
Oct 19 02:00:39 [fra-platform](https://my.papertrailapp.com/systems/fra-platform/events?focus=1651586122248204377&selected=1651586122248204377) [app/postgres.2488041](https://my.papertrailapp.com/events?q=program%3Aapp%2Fpostgres.2488041&focus=1651586122248204377&selected=1651586122248204377) [PURPLE] [57-3]  sql_error_code = 53200 time_ms = "2023-10-19 09:00:38.802 UTC" pid="1331187" proc_start_time="2023-10-19 09:00:02 UTC" session_id="6530f012.144ff3" vtid="181/141" tid="41447599" log_line="23" database="d5hkm0brvqv0u2" connection_source="[3.254.201.206](https://my.papertrailapp.com/systems/fra-platform/events?q=%223.254.201.206%22&focus=1651586122248204377&selected=1651586122248204377)(56800)" user="u19b6eg7mpjl5" application_name="[unknown]" STATEMENT:  refresh materialized view assessment_fra_2025."activity_log_SWE";
Oct 19 02:00:39 [fra-platform](https://my.papertrailapp.com/systems/fra-platform/events?focus=1651586122248204378&selected=1651586122248204378) [app/postgres.2488079](https://my.papertrailapp.com/events?q=program%3Aapp%2Fpostgres.2488079&focus=1651586122248204378&selected=1651586122248204378) [PURPLE] [57-1]  sql_error_code = 53200 time_ms = "2023-10-19 09:00:38.802 UTC" pid="1331225" proc_start_time="2023-10-19 09:00:02 UTC" session_id="6530f012.145019" vtid="218/137" tid="41447600" log_line="21" database="d5hkm0brvqv0u2" connection_source="[3.254.201.206](https://my.papertrailapp.com/systems/fra-platform/events?q=%223.254.201.206%22&focus=1651586122248204378&selected=1651586122248204378)(57160)" user="u19b6eg7mpjl5" application_name="[unknown]" ERROR:  out of memory
```

Platform error:

Image

### PR DESCRIPTION
**Pull request by sorja**
Change:
Don't fetch latest changes from public.activity_log
Update all views

Done:
Test on review before merging --> No out of memory errors and platform performance wasn't hindered when running the refresh

### CODE REVIEW

File: src/server/schedulers/initCountryActivityLog.ts [hunk 1]
```diff
@@ -1,53 +1,33 @@
...
-      )
-
-      await Promise.all(
-        newActivityLogEntries.map(async (entry) => {
-          const { assessmentUuid, cycleUuid, countryIso } = entry
-          const assessment = assessments.find((assessment) => assessment.uuid === assessmentUuid)
-          const cycle = assessment?.cycles.find((cycle) => cycle.uuid === cycleUuid)
-
-          if (assessment && cycle && countryIso) {
+      await Promises.each(assessments, (assessment) =>
```
Comment (minotogna) [inline src/server/schedulers/initCountryActivityLog.ts]: does this refresh all countries anyway? would it be more efficient to refresh only when data has been updated ?

Comment (sorja) [reply src/server/schedulers/initCountryActivityLog.ts]: Do you mean we would query the `public.activity_log` then anyway?
Eg. return the previous condition?

Comment (minotogna) [reply src/server/schedulers/initCountryActivityLog.ts]: I would say so, I commented about the query, and the timestamp used as comparison .
or why updating all countries now?
what do you think?

**Comment by sorja** [comment]
Only one scheduler
First run country summary
and then with condition we run activity log

Same iteration when going through cycles

File: src/server/schedulers/initMaterializedViews.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          // 1. refresh country summary
+          await AreaController.refreshSummaries({ assessment, cycle })
+          Logger.info(`[${name}:CountrySummary] ${assessment.props.name} ${cycle.name} refreshed`)
+
+          // 2. refresh countries activity log
+          const countryISOs = entries[assessment.uuid]?.[cycle.uuid] ?? []
+          await Promises.each(countryISOs, async (countryIso) => {
+            Logger.debug(`[${name}:CountryActivityLog] ${assessment.props.name} ${cycle.name} refreshing`)
+            await CountryActivityLogRepository.refreshMaterializedView({ assessment, cycle, countryIso })
+            Logger.info(`[${name}:CountryActivityLog] ${assessment.props.name} ${cycle.name} refreshed`)
```
Comment (sorja) [inline src/server/schedulers/initMaterializedViews.ts:63]: Separate function for 1 and 2 for readability?

eg

```suggestion
_refreshSummer(assessment, cycle)
_refreshCountryActivityLog(countryISOs, assessment, cycle)
```

Comment (minotogna) [reply src/server/schedulers/initMaterializedViews.ts:63]: Open the file , it's not that bad . We can do that in the future if needed . What do you think ?

**Review by minotogna — APPROVED** [review]

**Pull request merged**